src/enum_net.cpp

```diff
@@ -363,6 +363,13 @@ namespace {
 			switch(rt_attr->rta_type)
 			{
 			case IFA_ADDRESS:
+				// if this is a point-to-point link then IFA_LOCAL holds
+				// the local address while IFA_ADDRESS is the destination
+				// don't overwrite the former with the latter
+				if (!ip_info->interface_address.is_unspecified())
```

Comment:
the only thing that makes me a little bit nervous is that this function now has a new (subtle) precondition that this `interface_address` field be in a default constructed state or cleared somehow.
Perhaps we should just clear this field at the top of the function, to simplify the interface.